File: C++/Medium/Accepted/count-triplets-that-can-form-two-arrays-of-equal-xor_8.cpp
class Solution {
public:
    int countTriplets(vector<int>& arr) {
        
        int n = arr.size();
        vector<int> pre(n+1);
        for (int i = 0; i < n; i++) pre[i+1] = pre[i] ^ arr[i];
        int ans = 0;
        for (int i = 0; i < n; i++)
            for (int j = i + 1; j < n; j++)
                for (int k = j; k < n; k++)
                    if ((pre[j] ^ pre[i]) == (pre[k+1] ^ pre[j])) ans++;
        return ans;
    }
};

File: Rust/Medium/CompileError/count-triplets-that-can-form-two-arrays-of-equal-xor_8.rs
impl Solution {
    pub fn count_triplets(arr: Vec<i32>) -> i32 {
        
        let mut res = 0;
        let mut xor = 0;
        let mut count = vec![vec![0; arr.len()]; arr.len()];
        for i in 0..arr.len() {
            xor ^= arr[i];
            for j in i + 1..arr.len() {
                if xor == (xor ^ arr[j]) {
                    res += j - i;
                }
                count[i][j] = res;
            }
            res = 0;
        }
        res = 0;
        for i in 0..arr.len() {
            for j in i + 1..arr.len() {
                for k in j..arr.len() {
                    if arr[i] == arr[k + 1] {
                        res += 1;
                    }
                }
            }
        }
        res
    }
}